tests/integration/api/v1/curated_recommendations/test_curated_recommendations.py

```diff
@@ -374,39 +373,171 @@ async def test_curated_recommendations_topics(self, topics):
+                Topic.TRAVEL,
+                Topic.FOOD,
+                Topic.ARTS,
+                Topic.POLITICS,
+                Topic.GAMING,
+                Topic.SPORTS,
+                Topic.SCIENCE,
+                Topic.SELF_IMPROVEMENT,
+                Topic.PARENTING,
+            ],
+            [
+                Topic.EDUCATION,
+                Topic.PERSONAL_FINANCE,
+                Topic.BUSINESS,
+                Topic.TECHNOLOGY,
+                Topic.TRAVEL,
+                Topic.FOOD,
+                Topic.ARTS,
+                Topic.POLITICS,
+                Topic.GAMING,
+                Topic.SPORTS,
+                Topic.SCIENCE,
+                Topic.SELF_IMPROVEMENT,
+                Topic.PARENTING,
+                Topic.CAREER,
+            ],
+            [
+                Topic.EDUCATION,
+                Topic.PERSONAL_FINANCE,
+                Topic.BUSINESS,
+                Topic.TECHNOLOGY,
+                Topic.TRAVEL,
+                Topic.FOOD,
+                Topic.ARTS,
+                Topic.POLITICS,
+                Topic.GAMING,
+                Topic.SPORTS,
+                Topic.SCIENCE,
+                Topic.SELF_IMPROVEMENT,
+                Topic.PARENTING,
+                Topic.CAREER,
+                Topic.HEALTH_FITNESS,
+            ],
+        ],
+    )
+    async def test_curated_recommendations_preferred_topic(self, preferred_topics):
+        """Test the curated recommendations endpoint accepts 1-15 preferred topics &
+        top N recommendations contain the preferred topics.
+        """
         async with AsyncClient(app=app, base_url="http://test") as ac:
             response = await ac.post(
-                "/api/v1/curated-recommendations", json={"locale": "en-US", "topics": ["health"]}
+                "/api/v1/curated-recommendations",
+                json={"locale": "en-US", "topics": preferred_topics},
             )
             data = response.json()
             corpus_items = data["data"]
 
             assert response.status_code == 200
             # assert total of 80 items returned
             assert len(corpus_items) == 80
-            # assert boost_preferred_topic was called
-            boost_preferred_topic_spy.assert_called_once()
 
-            # extract IDs from both lists
-            fixture_ids = [
-                item["id"] for item in fixture_response_data["data"]["scheduledSurface"]["items"]
-            ]
-            corpus_ids = [item["scheduledCorpusItemId"] for item in corpus_items]
-            # assert that recs were reordered
-            assert fixture_ids != corpus_ids
+            # get the top N recs
+            top_recs = min(10, 2 * len(preferred_topics))
+            # store the topics in the top N recs in an array
+            top_topics = [item["topic"] for item in corpus_items[:top_recs]]
+            # assert the intersection of the two sets
+            assert set(preferred_topics) & set(top_topics)
```

Comment:
```suggestion
This asserts that at least one of the `top_topics` is preferred. Instead, we want to assert that _all_ `top_topics` are preferred:
```suggestion
            # assert that the first N number of recommendations are all preferred
            assert all([item["topic"] in preferred_topics for item in corpus_items[:top_recs]])
``````

Reply:
@mmiermans that won’t be possible, and wouldn’t be the case since there is 15 topics & only 10 slots, right ?